### DISCUSSION TITLE
changeProfilePicture not working

### DISCUSSION THREAD

**Discussion by hthoene** [Auties00/WhatsappWeb4j#238] [General]
Hello, is the method changeProfilePicture already working? It seems like it is not working for me.
I use it like this:

`byte[] fileContent = Files.readAllBytes(new File(App.resultPath).toPath());`
`api.changeProfilePicture(fileContent).join();`

**Comment by Auties00** [comment]
i'll look into it. Thanks for your report

**Reply by hthoene** [reply]
Thank you! If it is working for you, please let me know. Maybe I'm doing something wrong.

**Reply by Auties00** [reply]
> Thank you! If it is working for you, please let me know. Maybe I'm doing something wrong.

Can you make sure that the file you are trying to upload is a jpeg?

**Reply by hthoene** [reply]
I tried jpg and png with 640px resolution. Does it work for you?

**Reply by hthoene** [reply]
Does it matter if I use "file.jpg" or "file.jpeg"?

**Reply by Auties00** [reply]
no, it only matters that the file is encoded correctly

**Reply by hthoene** [reply]
Seems to be not working for me

**Reply by hthoene** [reply]
When I use null for the byte[] it works (it removes the image). The code that I'm using to get the bytes is:
`byte[] media = Files.readAllBytes(Path.of(App.testPath));`
I also tried:
```
BufferedImage bImage = ImageIO.read(new File(App.testPath));
ByteArrayOutputStream bos = new ByteArrayOutputStream();
ImageIO.write(bImage, "jpg", bos );
byte [] media = bos.toByteArray();
```

**Reply by Auties00** [reply]
Can confirm it was not working, fixed on master branch
If you use maven releases, I'll publish one tomorrow

**Reply by hthoene** [reply]
Thank you so much!
